_includes/ballot-nav.html

```diff
@@ -1,43 +1,52 @@
-      {% if referendums and referendums != empty %}
-        <div class="ballot-nav__group">
-          <h4>{% include referendum_measure_heading.html referendum_measure_display=locality.referendum_measure_display %}</h4>
-          {% for ballot_item in referendums %}
-              {% assign tabIndxVar = tabIndxVar | plus:1 %}
-              {% assign referendum = site.referendums | where: "path", ballot_item | first %}
-              <p {% if page.url == referendum.url %}class="current-page"{% endif %}>
-                <a href="{{ referendum.url | prepend: site.baseurl }}" tabindex="{{ tabIndxVar }}">
-                  Measure {% if referendum.number %}{{ referendum.number }} {% endif %}
-                  <span class="ballot-nav__measure-title">{{ referendum.title | smartify }}</span>
-                </a>
-              </p>
-          {% endfor %}
-        </div>
-      {% endif %}
-      {% if office_elections and office_elections != empty %}
-        {% for ballot_item in office_elections %}
-          {% assign tabIndxVar = tabIndxVar | plus:1 %}
-          <div class="ballot-nav__group">
-            {% if ballot_item.label %}
-              <h4>{{ ballot_item.label | escape }}</h4>
-            {% endif %}
-            {% for office_election_path in ballot_item.items %}
-            {% assign tabIndxVar = tabIndxVar | plus:1 %}
-              {% assign office_election = site.office_elections | where: "path", office_election_path | first %}
-              <p {% if page.url == office_election.url %}class="current-page"{% endif %}>
-                <a href="{{ office_election.url | prepend: site.baseurl }}" tabindex="{{ tabIndxVar}}">{{ office_election.title }}</a>
-              </p>
-            {% endfor %}
-          </div>
-        {% endfor %}
+  <h3 class="ballot-nav__locality-heading">{{ locality.name | escape }}</h3>
+  <div class="ballot-nav__group">
+    <h3 class="ballot-nav__section-heading">Upcoming Races</h3>
+  </div>
+  <section>
+    {% for ballot in ballots %}
+    {% assign referendums = ballot.referendums %}
+    {% assign office_elections = ballot.office_elections %}
+    <h3 class="ballot-nav__date-heading ballot-nav__group">{{ ballot.election | date: "%B %-d, %Y" }}</h3>
+    {% if referendums and referendums != empty %}
+    <div class="ballot-nav__group">
+      <h4 class="ballot-nav__section-heading">{% include referendum_measure_heading.html referendum_measure_display=locality.referendum_measure_display %}</h4>
+      <ul>
+      {% for ballot_item in referendums %}
+      {% assign tabIndxVar = tabIndxVar | plus:1 %}
+      {% assign referendum = site.referendums | where: "path", ballot_item | first %}
+      <li class="ballot-nav__list-item{% if page.url == referendum.url %} current-page{% endif %}">
+        <a href="{{ referendum.url | prepend: site.baseurl }}" tabindex="{{ tabIndxVar }}">
+          Measure {% if referendum.number %}{{ referendum.number }} {% endif %}
```

Comment:
It looks like there aren't any referendums for the November election, but the nav is showing one, without a number (which is right, since there isn't one):
<img width="431" alt="Screen Shot 2020-02-09 at 9 48 20 PM" src="https://user-images.githubusercontent.com/28023047/74124340-ce79a180-4b86-11ea-8311-aed0f8d10b6e.png">

Maybe the `{% if referendums and referendums != empty %}` check isn't correctly catching the referendum-free November ballot, or somehow the data has a placeholder or some other funky thing to make this item appear.

Reply:
Could it be because there are 2 ballot measure committees in the spreadsheet for the November election, but the measures haven't qualified for the ballot yet, so there are no ballot measure names listed in the spreadsheet for November?

Reply:
Oh, great insight! That might be it, if that's causing whatever is in the `referendum` variable to have content/not be empty for some reason. We'll have to dig into it a bit more.

Reply:
Yeah this seems like a likely culprit.

Another thing I noticed is that the backend builds a nameless referendum file for the 2020-11-03 election.
![Screen Shot 2020-02-10 at 10 59 26 AM](https://user-images.githubusercontent.com/20404311/74210483-d138cd00-4c40-11ea-9c87-54252c31bb72.png)

But then the frontend doesn't seem to pull this file when it pulls the data from the backend. How would it render an unnamed referendum if there's no unnamed referendum file there? 🤔
![Screen Shot 2020-02-10 at 11 57 16 AM](https://user-images.githubusercontent.com/20404311/74210511-e57cca00-4c40-11ea-9aac-c4680fc84343.png)

Reply:
I printed the item that Jekyll is handling when it renders that blank Measure. It seems like it's that nameless referendum file, but I don't see the file in the data. What data file could be pointing at that fictitious file and telling Jekyll to render it?
![Screen Shot 2020-02-10 at 8 15 47 PM](https://user-images.githubusercontent.com/20404311/74210954-a485b500-4c42-11ea-906a-16fcbf6f0f54.png)

Reply:
Bingo. There's a reference to the nonexistent, unnamed file in the _elections markdown file for the oakland/2020-11-03 election
![Screen Shot 2020-02-10 at 8 21 14 PM](https://user-images.githubusercontent.com/20404311/74211042-09410f80-4c43-11ea-9980-baa08546b9cf.png)

@sfdoran can we give that measure a placeholder short title and see if builds properly in the data?

Reply:
Sidenote: funny that trying to pull data from a missing file doesn't cause Jekyll to crash or even to complain. Is there something we can do to make Jekyll alert us if data is missing in the future?

Reply:
Awesome, thanks for looking into it! A placeholder would be nice, to see if the data shows up on the site. 

Yea, that's really weird that the file isn't there but Jekyll seems to think something is...